src/main/java/com/minecolonies/core/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -159,26 +166,25 @@ public CompoundTag serializeNBT()
      */
     public int getDepthLimit(final Level level)
     {
-        if (this.getBuildingLevel() == 1)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_1, level);
-        }
-        else if (this.getBuildingLevel() == 2)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_2, level);
-        }
-        else if (this.getBuildingLevel() == 3)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_3, level);
-        }
-        else if (this.getBuildingLevel() == 4)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_MAX, level);
-        }
-        else
+        int buildingY = this.getLadderLocation().getY() - 5;
+
+        int buildingLevels = getBuildingLevel();
+        int yLevel = 0;
```

Comment:
shouldnt this be 48?

Reply:
initial doesnt matter since it always takes the first one that is lower

Reply:
Unless `miningLevel < buildingY` is never true.  Which could be the case if the max depth is increased again at some point and the mine is built low enough.

It also returns this initial value directly at building level 0, which seems like a thing that could more easily happen.